Source/WebCore/page/Navigator.cpp

```diff
@@ -137,15 +137,25 @@ bool Navigator::canShare(Document& document, const ShareData& data)
     if (!document.isFullyActive() || !validateWebSharePolicy(document))
         return false;
 
-    bool hasShareableTitleOrText = !data.title.isNull() || !data.text.isNull();
-    bool hasShareableURL = !!shareableURLForShareData(document, data);
 #if ENABLE(FILE_SHARE)
     bool hasShareableFiles = document.settings().webShareFileAPIEnabled() && !data.files.isEmpty();
 #else
     bool hasShareableFiles = false;
 #endif
 
-    return hasShareableTitleOrText || hasShareableURL || hasShareableFiles;
+    if (data.title.isNull() && data.text.isNull() && data.url.isNull() && !hasShareableFiles)
+        return false;
+
+#if ENABLE(FILE_SHARE)
+    bool hasTitleTextOrURL = !data.title.isNull() || !data.text.isNull() || !data.url.isNull();
+    if (!hasTitleTextOrURL && !hasShareableFiles)
```

Comment:
Isn't that redundant with line 146 check?
And why is this check controlled by #if ENABLE(FILE_SHARE)?

Reply:
yeah, true... I was coding it to match the spec text. I'll fold this up.